proposals/devcontainer-id-variable.md

```diff
@@ -0,0 +1,60 @@
+## Goal
+
+Allow features to refer to an identifier that is unique to the dev container they are installed into and that is stable across rebuilds.
+
+E.g., the `docker-in-docker` feature needs a way to mount a volume per dev container.
+
+## Proposal
+
+The identifier will be referred to as `${devcontainerId}` in the feature metadata and that will be replaced with the dev container's id. It should only be used in parts of the metadata that is not used for building the image because that would otherwise prevent pre-building the image at a time when the dev container's id is not known yet. Excluding boolean and enum properties the properties supporting `${devcontainerId}` are: `entrypoint`, `mounts`, `customizations`.
```

Comment:
Does it make sense to allow `devcontainerId` to be passed into lifecycle hooks in some way? I'm not aware of a need for this currently, so we could always defer it and add it later once there is a know use case.

Reply:
That makes sense. (As discussed before: The only limitation is that it is not available when building the image because that can be prebuilt in CI.)